test/components/connectors/test_openapi_service.py

```diff
@@ -126,19 +123,62 @@ def test_run(self, openapi_mock, test_files_path):
             ]
         )
         messages = [ChatMessage.from_assistant(mock_message)]
+        mock_return_value = Mock(_raw_data={"commits": 1, "files": 2, "lines": 3})
+        with patch.object(OpenAPIServiceConnector, "_invoke_method", return_value=mock_return_value) as mock_invoke:
+            result = connector.run(messages=messages, service_openapi_spec=spec, service_credentials="fake_key")
+            assert result["service_response"][0].content == '{"commits": 1, "files": 2, "lines": 3}'
```

Comment:
This is no good, let's not mock the component we're testing. We should mock the OpenAPI object.
I've done something similar in [this test](https://github.com/deepset-ai/haystack/blob/f3be576b5c30a2c79c39cb1ad66606cbd7edec1f/test/components/connectors/test_openapi_service.py#L110-L144).

Reply:
Ok, will do, thanks for your input